src/gemm_v2.cpp

```diff
@@ -324,6 +326,173 @@ inline void ProfilingRecordStop(const Handle& handle, HipEventPtr& start, HipEve
     handle.ResetKernelTime();
     handle.AccumKernelTime(mS);
 }
+
+#define HIP_BLAS_STATUS_CHECK(cmd)                       \
+    do                                                   \
+    {                                                    \
+        hipblasStatus_t e = cmd;                         \
+        if(e != hipblasStatus_t::HIPBLAS_STATUS_SUCCESS) \
+        {                                                \
+            return e;                                    \
+        }                                                \
+    } while(0)
+
+template <typename DataTypeAB, hipDataType HipTypeAB, typename DataTypeC, hipDataType HipTypeC>
```

Comment:
I don't think that `HipTypeAB` should be a template parameter. It bloats the binary instantiating unique function for each distinct type.